anvio/docs/artifacts/groups-txt.md

```diff
@@ -1,13 +1,15 @@
-This is a two-column, TAB-delimited file that describes which samples belong to which groups. It is a required input for computing enrichment scores of metabolic modules in %(anvi-compute-functional-enrichment)s.
+This is a 2-column TAB-delimited text file to associate a given set of items with a set of groups. Depending on the context, items here may be individual samples or genomes. The first column can have a header name `item`, `sample`, `genome` or anything else that is appropriate, and list the items that are relevant to your input data. The second column should have the header `group`, and associate each item in your data with a group.
```

Comment:
@ivagljiva, I broadened the description of the groups-txt file. Is it OK?

Reply:
Yes, I think it is okay, but in that case we should probably change which headers are accepted for this file. 
Right now, for running `anvi-compute-functional-enrichment` on metabolic modules, the first column is required to have the header `sample` - if not, you get an error that looks like this:
```
Config Error: The file 'test_diff_groups_txt_header.txt' does not contain the right type of
              header. It was expected to have these: 'sample, group', however it had these:
              'group'
```
(In this case, my file `test_diff_groups_txt_header` had `item` as the header of the first column)

I think it is likely we will use this groups-txt file for broader purposes in the future and it need not be restricted to that header, which is why I like that you changed this description. But now the code should also be flexible enough to accept any header. I can change it, if you agree.

Reply:
thank you very much for noticing this, @ivagljiva. I think it would be best if we changed the description. I think the best way to do is to have a function in `utils.py`, parallel to `get_samples_txt_file_as_dict`.

That function could do the sanity check for 'groups-txt', and look for a `group` column regardless of the number of columns in a given tab delimited file to only return the first and the groups column information (or it could also return a tuple of two dictionaries where in one of them keys are items and values are groups, and in the other keys are groups and values are a list of items in each group. With a function like that in utils, this entire code in `genomedescriptions.py`,

``` python
   (...)
            if self.layer_groups_input_file_path:
                filesnpaths.is_file_tab_delimited(self.layer_groups_input_file_path,)
                self.layer_groups = {}

                if 'group' not in utils.get_columns_of_TAB_delim_file(self.layer_groups_input_file_path):
                    raise ConfigError("A groups-txt file should have a single column that is called `group`.")

                layer_groups_d = utils.get_TAB_delimited_file_as_dictionary(self.layer_groups_input_file_path)

                for layer_name in layer_groups_d:
                    group_name = layer_groups_d[layer_name]['group']

                    if not group_name in self.layer_groups:
                        self.layer_groups[group_name] = []

                    self.layer_groups[group_name].append(layer_name)
            elif layer_groups:
                self.layer_groups = layer_groups

            # Finally, last AND least, a small helper dictionary we will use if there are gorups defined
            # by the user:
            if self.layer_groups:
                self.layer_groups_defined = True
                for group_name in self.layer_groups:
                    for layer_name in self.layer_groups[group_name]:
                        self.layer_name_to_group_name[layer_name] = group_name

   (...)
```

to this one to be used from everywhere in similar ways:

``` python
    (..)
    self.layer_groups, self.layer_name_to_group_name = utils.get_samples_txt_file_as_dicts(file_path)
    (...)
```

If you have the energy and throughput, please feel free to do it. If not, let me know and I can do it myself :)